docs/user-guide.md

```diff
@@ -0,0 +1,223 @@
+# OpenShift Appliance User Guide
+
+## What is OpenShift Appliance
+The `openshift-appliance` is a container-based utility that builds a disk image that facilitates the installation of OpenShift with an [Agent-based installer](https://cloud.redhat.com/blog/meet-the-new-agent-based-openshift-installer-1).
+
+Its main purpose is to enable a completely offline installation of an OpenShift cluster, Thus, all required images are included in the appliance disk image.
+
+
+## Download
+OpenShift Appliance is available for download at: https://quay.io/edge-infrastructure/openshift-appliance
+
+## High-Level Flow Overview
+
+![hl-overview.png](images%2Fhl-overview.png)
+
+### Lab
+* This is where `openshift-appliance` gets used to create a raw space disk image.
+  * **raw:** so it can be copied as-is to multiple servers. 
+  * **sparse:** to keep the size as small as possible.
+* The end result is a generic disk image with a partition layout as follows:
+  ``` bash
+  Name       Type        VFS      Label       Size  Parent
+  /dev/sda2  filesystem  vfat     EFI-SYSTEM  127M  -
+  /dev/sda3  filesystem  ext4     boot        350M  -
+  /dev/sda4  filesystem  xfs      root        180G  -
+  /dev/sda5  filesystem  ext4     agentboot   1.2G  -
+  /dev/sda6  filesystem  iso9660  agentdata   18G   -
+  ```
+* The two additional partitions:
+  * `agentboot`: Agent-based installer ISO:
+    * Allows a first boot.
+    * Used as a recovery / re-install partition (with an added grub manu entry).
+  * `agentdata`: OCP release images payload.
+* Note that sizes may change, depending on the configured `diskSizeGB` and the selected OpenShift version configured in `appliance-config.yaml`
```

Comment:
the sizes of partitions may change? is that based on percentage? what is the percentage then?

Reply:
Can you elaborate? what information is mentioned twice and what is missing here?

Reply:
So the sizes of 'agentboot' and 'agentdata' are predefined on build (these partitions are read-only). We calculate the required size according the the boot ISO and release images. The 'root' partition is taking the remaining space on the disk (according to 'diskSizeGB'). Not sure we want a long explanation here in the guide with these details, maybe we could extract it to some 'architecture' doc.

Reply:
Look for "sizes may". Same info pretty much.